doc/sfp-cmis/cmis-init.md

```diff
@@ -0,0 +1,424 @@
+support in SONiC.
+
+The Common Management Interface Specification (CMIS) provides a variety of features
+and support for different transceiver form factors. A CMIS transceiver may support
+multiple application, and the application initialization sequence is now mandatory
+upon port mode changes. Otherwise the link will be down if the host port mode
+does not match the selected application on the CMIS transceiver.
+
+The feature is built on top of SONiC **sfp-refactor** framework to provide a
+platform-independent solution, and the individual platforms could easily enable
+this feaure by having its **Sfp** object inherited from **SfpOptoeBase**.
+
+**Example:**  
+```
+from sonic_platform_base.sonic_xcvr.sfp_optoe_base import SfpOptoeBase
+
+class Sfp(SfpOptoeBase):
+
+    def __init__(self, sfp_index):
+        SfpOptoeBase.__init__(self)
+        self.index = sfp_index
+
+    def get_port_type(self):
+        return self.SFP_PORT_TYPE_QSFPDD
+
+    def get_eeprom_path(self):
+        # platform-specific per-port I2C bus
+        bus_id = 32 + self.index
+        return "/sys/bus/i2c/devices/{}-0050/eeprom".format(bus_id)
+```
+
+This feature could also be disabled by the platform-specific **pmon_daemon_control.json**
+
+**Example:**  
+```
+$ cat sonic-buildimage/device/VENDOR/PLATFORM/pmon_daemon_control.json
+{
+    "skip_xcvrd_cmis_manager": true
+}
+```
+
+The scope of this feature are as follows:
+
+- **CMIS software initialization for the default application.**  
+  - All the lanes of the CMIS module will be reconfigured to use the default application.
+  - Only staged control set 0 will be supported
+  - No speed negotiation.
+  - No custom signal integrity settings
+  - Implement the procedures defined in Appendix D.1.3 and D.2.2 of [CMIS 5](http://www.qsfp-dd.com/wp-content/uploads/2021/05/CMIS5p0.pdf)
+- **sonic-platform-common**: Update **sonic-xcvr** for CMIS application advertising and initialization
+- **sonic-platform-daemons**: Update **sonic-xcvrd** for state-based CMIS application initialization
+to support multiple CMIS transceivers in one single thread
+- **sonic-utilities**: Update the **sfputil** and **sfpshow** for CMIS application advertisement
+
+## Functional Requirements
+
+1. Ability to parse the advertised applications from the transceivers
+2. Ability to post the advertised applications to the STATE_DB
+3. Ability to support multiple CMIS transceivers in one single thread.
+4. Ability to detect the errors of the application initialization on the transceivers
+
+**Note:**  
+The duration of the CMIS application initialization greatly differs from transceivers
+to transceivers, while some take 3 seconds for activating the 4x100G mode, some take 15 seconds.
+
+## Warm Boot Requirements
+
+Functionality should continue to work across warm boot. 
+- The CMIS application initialization should not be performed during WarmBoot.
+- The CMIS application initialization should be skipped if no application code updates.
```

Comment:
No, it's not necessary, the CmisMamagerTask will always double-check the current application mode against the desired application from the CONFIG_DB, hence this will be processed when the XCVR INSERTION events are replayed after fast-reboot.